From: Liam Howlett <liam.howlett@oracle.com>
Subject: test_maple_tree: add test for spanning store to most of the tree
Date: Wed, 29 Jun 2022 15:23:51 +0000

Test spanning almost the entire tree to detect if the root is dead and the
node is placed in the wrong location.

Link: https://lkml.kernel.org/r/20220629152340.3451959-3-Liam.Howlett@oracle.com
Fixes: ("test_maple_tree: Add test for spanning store to most of the tree")
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/test_maple_tree.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/lib/test_maple_tree.c~lib-test_maple_tree-add-testing-for-maple-tree-fix-2
+++ a/lib/test_maple_tree.c
@@ -35685,7 +35685,6 @@ static noinline void check_spanning_writ
 	mas_set(&mas, 1205);
 	MT_BUG_ON(mt, mas_walk(&mas) != NULL);
 	mtree_unlock(mt);
-	mt_dump(mt);
 	mt_validate(mt);
 	mtree_destroy(mt);
 
@@ -35817,6 +35816,15 @@ static noinline void check_spanning_writ
 	MT_BUG_ON(mt, mas_walk(&mas) != NULL);
 	mtree_unlock(mt);
 	mtree_destroy(mt);
+
+	mt_init_flags(mt, MT_FLAGS_ALLOC_RANGE);
+	for (i = 0; i <= 100; i++)
+		mtree_test_store_range(mt, i * 10, i * 10 + 5, &i);
+
+	mtree_lock(mt);
+	mas_set_range(&mas, 76, 875);
+	mas_store_gfp(&mas, NULL, GFP_KERNEL);
+	mtree_unlock(mt);
 }
 
 static noinline void check_null_expand(struct maple_tree *mt)
_
